evap/evaluation/management/commands/ts.py

```diff
@@ -0,0 +1,70 @@
+import argparse
+import os
+import subprocess  # nosec
+import sys
+import unittest
+
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from django.test.runner import DiscoverRunner
+
+
+class ExportRunner(DiscoverRunner):
+    test_loader = unittest.TestLoader()
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.test_loader.testMethodPrefix = 'generate_fixtures'
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser: argparse.ArgumentParser):
+        subparsers = parser.add_subparsers(dest='command', required=True)
+        compile_parser = subparsers.add_parser('compile')
+        compile_parser.add_argument(
+            '--watch', action='store_true',
+            help='Watch scripts and recompile when they change.',
+        )
+        subparsers.add_parser('test')
+        subparsers.add_parser('generate_fixtures')
+
+    def handle(self, *args, **options):
+        if options['command'] == 'compile':
+            self.compile(**options)
+        elif options['command'] == 'test':
+            self.test(**options)
+        elif options['command'] == 'generate_fixtures':
+            self.generate_fixtures(**options)
+
+    def run_command(self, command):
+        try:
+            subprocess.run(command, check=True)  # nosec
+        except FileNotFoundError:
+            print(f'Could not find {command[0]} command', file=self.stderr)
+        except KeyboardInterrupt:
+            pass
+        except subprocess.CalledProcessError as error:
+            sys.exit(error.returncode)
+
+    def compile(self, watch, **_options):
+        static_directory = settings.STATICFILES_DIRS[0]
+        command = [
+            'npx',
+            'tsc',
+            '--project',
+            os.path.join(static_directory, 'ts', 'tsconfig.compile.json'),
+        ]
+
+        if watch:
+            command += ['--watch']
+
+        self.run_command(command)
+
+    def test(self, **_options):
+        self.run_command(['npx', 'jest'])
```

Comment:
I'd say, ideally, `generate_fixtures` and `compile` should realize that they have no work to do and just immediately return.

If they can't do that, I'd add a manual cheap check (so very rarely, a corrupted fixture might not be detected) and add a --force-build parameter to be used in this case.

Reply:
Implementing said check for `generate_fixtures` sounds hard (or at least I have no idea). Technically, it would need to detect that all Django files (Python and templates) haven’t changed since the last run (or, even more sophisticated and complicated, only the subset of Django files which were needed for the rendered pages). I don’t know whether using [git via python](https://stackoverflow.com/questions/33733453/get-changed-files-using-gitpython) would be helpful here

Reply:
Hmm, when I wrote my comment, I was thinking something along the lines of: glob all files in the source dir and hash them. If the has equals some stored value, don't compile. Otherwise, compile and store the hash. [Could be done in a handful of lines in python](https://stackoverflow.com/a/36099389/12345551).

Alterantively, `make` has some similar features (using the last-changed-timestamp). Could we somehow utilize that?

Keep in mind, all of this is only a workaround for the case that these tools do not support incremental builds and are annoyingly slow.

Reply:
Okay, currently `management.py ts test` takes around 50s on my machine, 30s are consumed by jest (so 20s for `ts compile` + `ts render_pages`). For the small count of unit and frontend tests, this is rather long, but absolutely speaking, it is acceptable for me. If developers are getting tired of long runtimes, we can look more into it, but I don’t want to slow this PR down by looking into speeding things up (re: premature optimizations)